tiledb/sm/query/readers/aggregators/query_channel.h

```diff
@@ -85,6 +89,58 @@ class QueryChannel {
   ChannelAggregates aggregates_;
 };
 
+/* forward declaration */
+class Query;
+
+/**
+ * Replacement for the current QueryChannel, which does not work for more than
+ * the initial case with a default channel and only simple aggregates.
+ *
+ * Responsibility for choosing channel identifiers is the responsibility of
+ * `class Query`; this class merely carries the resulting identifier.
+ */
+class QueryChannelActual {
```

Comment:
Maybe we should rename this to QueryChannel and rename the one above to QueryChannelSerialization?

Reply:
This one: `class QueryChannel`
The previous one: `class LegacyQueryAggregates`

The existing class isn't really about channels so much as all the aggregates over the default channel. `class LegacyQueryAggregatesOverDefault` would be more accurate, but far wordier.

Reply:
The wordier version might be fine. It's only used in a few places.